File: frontend/src/types/prediction.ts
export interface PredictionRequest {
  location: string;
  carpet_area_sqft: number;
  floor_num: number;
  bathroom: number;
  balcony: number;
  furnishing: string;
  transaction: string;
  ownership: string;
  facing: string;
}

export interface PredictionResponse {
  predicted_price: number;
}

export const FURNISHING_OPTIONS = ["Unfurnished", "Semi-Furnished", "Furnished"] as const;
export const TRANSACTION_OPTIONS = ["Resale", "New Property", "Other", "Rent/Lease"] as const;
export const OWNERSHIP_OPTIONS = ["Freehold", "Co-operative Society", "Power Of Attorney", "Leasehold"] as const;
export const FACING_OPTIONS = [
  "East", "West", "North", "South",
  "North - East", "North - West", "South - East", "South -West",
] as const;

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class PredictionRequest(BaseModel):
    location: str
    carpet_area_sqft: float = Field(gt=0)
    floor_num: int
    bathroom: int = Field(ge=0)
    balcony: int = Field(ge=0)
    furnishing: str
    transaction: str
    ownership: str
    facing: str


class PredictionResponse(BaseModel):
    predicted_price: float